File: programmers/level2/기능개발.py
"""
문제 링크: https://programmers.co.kr/learn/courses/30/lessons/42586
문제 접근 방법: 단순 구현
"""

def solution(progresses, speeds):
    answer = []
    
    date = 0
    i = 0
    day = 0

    while i < len(progresses):
        progress = progresses[i]
        speed = speeds[i]
        
        if progress + date * speed >= 100:
            i+=1
            day += 1
        elif day > 0:
            answer.append(day)
            day = 0
            date+=1
        else:
            date+=1
    answer.append(day)
            
    return answer

File: programmers/level2/기능개발.java
import java.util.*;
class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        
        List<Integer> ans = new ArrayList<>();
        
        int date = 0;
        int i =0;
        int day = 0;
        
        while (i < progresses.length) {
            int progress = progresses[i];
            int speed = speeds[i];
            
            if(progress + date * speed >= 100) {
                i ++;
                day ++;
            } else if ( day > 0) {
                ans.add(day);
                day = 0;
                date ++;
            } else {
                date ++;
            }
        }
        ans.add(day);
        
        
        
        int[] answer = new int[ans.size()];
        for(int j=0; j< ans.size() ; j++ ){
            answer[j] = ans.get(i);
        }
        
        return answer;
    }
}